spec/bundler/plugin_spec.rb

```diff
@@ -251,14 +251,29 @@
     end
   end
 
+  describe "#add_hook" do
+    it "raises an ArgumentError on an unregistered event" do
+      ran = false
+      expect do
+        Plugin.add_hook("unregistered-hook") { ran = true }
+      end.to raise_error(ArgumentError)
+      expect(ran).to be(false)
+    end
+  end
+
   describe "#hook" do
     before do
       path = lib_path("foo-plugin")
       build_lib "foo-plugin", :path => path do |s|
         s.write "plugins.rb", code
       end
 
-      allow(index).to receive(:hook_plugins).with(event).
+      Bundler::Plugin::Events.send(:define, :EVENT_1, "event-1")
```

Comment:
I think in a spec_helper we may want to reset the set of `events` so there's no possibility for test ordering dependence